static/app/components/feedback/widget/feedbackWidget.tsx

```diff
@@ -1,38 +1,28 @@
-import {getCurrentHub} from '@sentry/react';
+import {useEffect} from 'react';
+import {BrowserClient, getCurrentHub} from '@sentry/react';
+import {Feedback} from '@sentry-internal/feedback';
 
 import ConfigStore from 'sentry/stores/configStore';
 import {useLegacyStore} from 'sentry/stores/useLegacyStore';
 
-import {FeedbackButton} from './feedbackButton';
-import {FeedbackModal} from './feedbackModal';
-
-interface FeedbackWidgetProps {
-  title?: string;
-  type?: string;
-}
-
 /**
- * The "Widget" connects the default Feedback button with the Feedback Modal
- *
- * XXX: this is temporary while we make this an SDK feature.
+ * Use this to display the Feedback widget in certain routes/components
  */
-export default function FeedbackWidget({
-  title = 'Report a Bug',
-  type,
-}: FeedbackWidgetProps) {
+export default function FeedbackWidget() {
   const config = useLegacyStore(ConfigStore);
+  const widgetTheme = config.theme === 'dark' ? 'dark' : 'light';
 
-  // Don't render anything if Sentry SDK is not already loaded
-  if (!getCurrentHub()) {
-    return null;
-  }
+  useEffect(() => {
+    const hub = getCurrentHub();
+    const client = hub && hub.getClient<BrowserClient>();
+    const feedback = client?.getIntegration(Feedback);
+    const widget = feedback?.createWidget({
+      colorScheme: widgetTheme,
+    });
+    return () => {
+      feedback?.removeWidget(widget);
+    };
+  }, [widgetTheme]);
 
-  const widgetTheme = config.theme === 'dark' ? 'dark' : 'light';
-  return (
-    <FeedbackModal title={title} type={type} widgetTheme={widgetTheme}>
-      {({open, showModal}) =>
-        open ? null : <FeedbackButton onClick={showModal} widgetTheme={widgetTheme} />
-      }
-    </FeedbackModal>
-  );
+  return null;
```

Comment:
@ryan953 would this be better as a hook?

Reply:
ya, it could be a hook because nothing is rendered.

just a name change and drop the `<` `>` in the callsite

Reply:
on second thought, it feels better to me as a component because we are "rendering" the widget (it's just outside of react vdom)

Reply:
if it's outside the vdom then it's a side-effect. We're mutating document.body, just like how hooks are used to mutate document.title and stuff like that.

Reply:
but does it matter? not a huge amount.